packages/playwright-test/src/reporters/junit.ts

```diff
@@ -133,6 +140,82 @@ class JUnitReporter implements Reporter {
     };
     entries.push(entry);
 
+    // Xray Test Management supports testcase level properties, where additional metadata may be provided
+    // some annotations are encoded as value attributes, other as cdata content; this implementation supports
+    // Xray JUnit extensions but it also agnostic, so other tools can also take advantage of this format
+    const properties: XMLEntry = {
+      name: 'properties',
+      children: [] as XMLEntry[]
+    };
+
+    if (this.embedAnnotationsAsProperties && test.annotations) {
+      for (const annotation of test.annotations) {
+        if (this.textContentAnnotations?.includes(annotation.type)) {
+          const property: XMLEntry = {
+            name: 'property',
+            attributes: {
+              name: annotation.type
+            },
+            text: annotation.description
+          };
+          properties.children?.push(property);
+        } else {
+          const property: XMLEntry = {
+            name: 'property',
+            attributes: {
+              name: annotation.type,
+              value: (annotation?.description ? annotation.description : '')
+            }
+          };
+          properties.children?.push(property);
+        }
+      }
+    }
+
+    // attachments are optionally embed as base64 encoded content on inner <item> elements
+    if (this.embedAttachmentsAsProperty) {
+      const evidence: XMLEntry = {
+        name: 'property',
+        attributes: {
+          name: this.embedAttachmentsAsProperty
+        },
+        children: [] as XMLEntry[]
+      };
+      for (const result of test.results) {
+        for (const attachment of result.attachments) {
+          let contents;
+          if (attachment.body) {
+            contents = attachment.body.toString('base64');
+          } else {
+            if (!attachment.path)
+              continue;
+            try {
+              const attachmentPath = path.relative(this.config.rootDir, attachment.path);
+              if (fs.existsSync(attachmentPath))
+                contents = fs.readFileSync(attachmentPath, { encoding: 'base64' });
```

Comment:
given the previous comment, where I mentioned it would make sense to add the warning to the `<system-err>` element, then I'll keep this approach and the `existsSync` as also exists in the other code block

Reply:
Sure, although this would require handling both "does not exist" and "error reading file" situations separately. In the latter case we can even print the error message to `<system-err>`.

Reply:
I have kept the same login as the existing oneon the initial code, where if file does not exist then it will add it to <system-err> (as it means the user provided a wrong path); if it fails reading the file, it is catch and doesn't do anything. 
This can be changed though, but I would say that then we would also need to change the existing logic on the other code.